code/game/gamemodes/game_mode.dm

```diff
@@ -88,6 +90,50 @@
 /datum/game_mode/proc/make_antag_chance(var/mob/living/carbon/human/character)
 	return
 
+///convert_roundtype()
+///Allows rounds to basically be "rerolled" should the initial premise fall through
+/datum/game_mode/proc/convert_roundtype()
+	var/list/datum/game_mode/runnable_modes = config.get_runnable_modes()
+	for(var/datum/game_mode/G in runnable_modes)
+		if(!G.reroll_friendly)	del(G)
+
+	if(!runnable_modes)	return 0
+
+	var/list/datum/game_mode/replacementmode = pickweight(runnable_modes)
+
+	switch(SSshuttle.emergency.mode) //Rounds on the verge of ending don't get new antags, they just run out
+		if(SHUTTLE_STRANDED, SHUTTLE_ESCAPE)
+			return 1
+		if(SHUTTLE_CALL)
+			if(SSshuttle.emergency.timeLeft(1) < initial(SSshuttle.emergencyCallTime)*0.5)
+				return 1
+
+	var/list/living_crew = list()
+
+	for(var/mob in player_list)
+		if(mob in living_mob_list)
+			living_crew += mob
+	if(living_crew.len / player_list.len <= 0.7) //If a lot of the player base died, we start fresh

```

Comment:
Wouldn't this also count "observe" ghosts and admin ghosts as dead players, or are they not included in player_list? And what about braindeads, (such as admin ghosted peoples)?

What exactly is included in player_list i guess? Because something like this should exclusively only count people on the manifest.